src/components/Chat.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+    <div class="chat-container">
+        <div class="close-button" @click="goToHome">X</div>
+        <h1>Chat with {{ chatWithUsername }}</h1>
+        <div class="messages-container" ref="messagesContainer">
+            <div v-for="(msg, index) in messages" :key="index" :class="msg.sender === 'You' ? 'sent' : 'received'">
+                <strong>{{ msg.sender }}:</strong> {{ msg.message }}
+            </div>
+        </div>
+        <div class="input-container">
+            <textarea v-model="newMessage" placeholder="Type your message"></textarea>
+            <button @click="sendMessage">Send</button>
+        </div>
+    </div>
+</template>
+
+<script>
+import io from 'socket.io-client';
+import { v4 as uuidv4 } from 'uuid';
+
+export default {
+    data() {
+        return {
+            messages: [],
+            newMessage: '',
+            socket: null,
+            chatWithUsername: ''
+        };
+    },
+    mounted() {
+        this.chatWithUsername = this.$route.params.username
+        if (!this.socket) {
+            this.socket = io('http://localhost:5000');
```

Comment:
Socket server IP is hard-coded, I don't think it should be like this (config is the best). Also see some console.log's if they deliberately there is no problem, otherwise LGTM.